packages/kernel-ui/src/context/PanelContext.tsx

```diff
@@ -60,35 +67,47 @@ export const PanelProvider: React.FC<{
     setPanelLogs([]);
   }, []);
 
-  const sendMessageWrapper: CallKernelMethod = useCallback(
-    async (payload) => {
-      if (isRequestInProgress.current) {
-        throw new Error('A request is already in progress');
+  const processRequests = useCallback(async (): Promise<void> => {
+    setIsLoading(true);
+    while (pendingRequests.current.length > 0) {
+      const request = pendingRequests.current.shift();
+      if (!request) {
+        break;
       }
```

Comment:
Would need to disable the lint rule but is IMO completely defensible in cases like this:
```suggestion
      const request = pendingRequests.current.shift()!;
```